validation/doc.go

```diff
@@ -0,0 +1,20 @@
+// Package validation contains logic to confirm that an OP Chain is either a valid Standard Chain
+// or a valid Frontier Chain. These checks are provided as Go tests.
+
+// It performs the following checks:
+
+// For Standard Chains:
+
+// * The Chain ID is unique.
+// * There is a valid, public RPC endpoint which returns a valid response.
+// * The resource metering config matches the Standard config.
+//   This includes the resource limit, elasticity multiplier, base fee maximum change denominator, minimum base fee, maximum base fee, and maximum gas.
+// * There is a valid gas price oracle, and the parameters (blob base fee scalar and base feed scalar) are within the permitted range.
+// * The L2 output oracle parameters (submission interval, block time and finalization period) all match the Standard config.
+// * The calculated genesis hash matches the hash declared in the chain’s config file.
+
+// For Frontier Chains:
+// * The Chain ID is unique.
+// * There is a valid, public RPC endpoint which returns a valid response.
```

Comment:
I'm slightly hesitant about committing a list of checks here, since it can easily get stale as we add or modify checks and forget to update these comments. But perhaps the advantages outweigh that?

I think in the longer term, doing https://github.com/ethereum-optimism/superchain-registry/issues/168 will help folks see at-a-glance what validation is being done.

If we do want to put this list in, then:

* I would suggest pulling out the uniqueness check into a "for all chains" section
* The "valid RPC endpoint" check is implicit at the moment. One of the other checks will fail if the RPC endpoint doesn't work.

Reply:
That's a fair concern! I added this comment because I spent about half an hour trying to understand what each check was doing. These checks are effectively part of our product surface area, so I can definitely imagine partners doing the same. 

I agree that #168 will help a lot once it's done. Two other alternatives in the interim:

1. We could add comments to the top of each file instead, so that it's more likely we'd remember to update them.
2. We could just make the tests _extremely_ clean and easy to read. I know this is fairly subjective but I think we could clean them up a little bit so that the things that are being checked are more obvious.

Reply:
Also:

> The "valid RPC endpoint" check is implicit at the moment. One of the other checks will fail if the RPC endpoint doesn't work.

Is this true also for Frontier Chains, which skip most of the other checks?

Reply:
> Is this true also for Frontier Chains, which skip most of the other checks?

Good question. I recently added this check which applies to all chains regardless of frontier/standard status https://github.com/ethereum-optimism/superchain-registry/pull/184.

Reply:
> I added this comment because I spent about half an hour trying to understand what each check was doing. These checks are effectively part of our product surface area, so I can definitely imagine partners doing the same.

Totally hear you on this. I like your option 2.

> We could just make the tests extremely clean and easy to read

I think there is a lot of scope for cleaning up the tests. I can work on this in tandem with https://github.com/ethereum-optimism/superchain-registry/issues/168. Also happy to make a new ticket to track this work, so it can be prioritised.

Reply:
@tessr we recently tidied up the declaration of the tests into a single file https://github.com/ethereum-optimism/superchain-registry/blob/main/validation/validation_test.go . Hopefully it is pretty clear which tests run on which class of chains now.